### DISCUSSION TITLE
Nts Geometry Conversion

### DISCUSSION THREAD

**Discussion by inforithmics** [Mapsui/Mapsui#1442] [Ideas]
As far as I understood is that there will be a parallel usage of NTS and Geometrie and I think I would be good to have methods to Convert from NTS to Geometry. So that they could be converted into each other.
If this methods would be already in Mapsui then it would be possible to already use NTS and convert it to Geometry and use some Features that are already in NTS. For example the NTS Shapefile Reader or NTS Algorithms like Simplifiers (for simplyfing Geometries) or Projections beyound the currently supported.
So what is the opinion. 
Would a Pull request be accepted that introduces Conversions from NTS to Geometry and Geometry to NTS? Should it be in in Mapsui.Nts Project or somewhere else?

**Comment by pauldendulk** [comment]
My initial idea was that there would be a longer period in which both were supported but I have done a lot of preparations and know the code better at this point, so right now I am hoping for a shorter period of overlap. An advantage of removing Geometry altogether is that we can move to MIT much quicker, and I now think this may be feasible in the upcoming major release (v4). So if we remove Geometries altogether in v4 the conversions methods will only be useful during a very short temporary solution, so I do not think this is worth the investment.

Then again, I do think it would be good if I accept some help in the migration to NTS. Perhaps just with the straight migration of Geometry layers to NTS layers. This will involve the same kind of knowledge. This will need some coordination between us. Also, I need to move some projects around before we can start. Mapsui.Core should be Moved back to Mapsui, and what is now in Mapsui should be moved to elsewhere. And before that the current PRs should be merged. It looks like I will have some time for Mapsui this Friday and weekend.

**Reply by janusw** [reply]
> My initial idea was that there would be a longer period in which both were supported but I have done a lot of preparations and know the code better at this point, so right now I am hoping for a shorter period of overlap. An advantage of removing Geometry altogether is that we can move to MIT much quicker, and I now think this may be feasible in the upcoming major release (v4).

+1 for that plan! Better make a clean cut once the NTS support is complete.

If anyone needs to continue working with the old structures, he could stick to the v3 branch, right? 

> Then again, I do think it would be good if I accept some help in the migration to NTS. Perhaps just with the straight migration of Geometry layers to NTS layers. This will involve the same kind of knowledge. This will need some coordination between us.

I am willing to help out here, if I can. Let's coordinate this in private ...

**Reply by pauldendulk** [reply]
@janusw Great to hear that!

**Comment by inforithmics** [comment]
Also +1 for directly migrating to NTS, currently I use mapsui with NTS (I need to convert from one to the other for certain functionality).

**Reply by pauldendulk** [reply]
Ah, so you already have conversion methods, that makes it a bit different. I keep the option i mind.

**Comment by pauldendulk** [comment]
I am about to merge a PR where Mapsui.Core is moved back into Mapsui. After this the actual migration of Mapsui.GeometryLayers to Mapsui.Nts can start.
